src/preprocess/LIDC-IDRI_preprocessing.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import argparse
+import numpy as np
+import nibabel as nib
+from tqdm import tqdm
+from pathlib import Path
+import SimpleITK as sitk
+
+import dicom2nifti
+
+
+def main(args):
+    input_dir = args.input_dir
+    output_dir = args.output_dir
+    
+    if output_dir.exists() is False:
+        output_dir.mkdir(parents=True)
+        
+    data_paths = sorted(list(input_dir.iterdir()))
+    for path in tqdm(data_paths):
+        itk_img = sitk.ReadImage(path.as_posix())
+        resampled_img = resample_to_istropic(itk_img)
+        
+        filename = path.parts[-1]
+        output_path = output_dir / filename
+        sitk.WriteImage(resampled_img, output_path.as_posix())
+
+
+def resample_to_istropic(itk_img):
+    w_res, h_res, d_res = itk_img.GetSpacing()[:]
+    w, h, d = itk_img.GetSize()[:]
+    
+    resized_height = h * h_res // 1.0
+    resized_width = w * w_res // 1.0
+    resized_depth = d * d_res // 1.0
+    target_shape = [resized_depth, resized_height, resized_width]
+    new_spacing = [1.0, 1.0, 1.0]
+
+    target_space = sitk.GetImageFromArray(np.ones(np.int32(list(target_shape) + [1]), dtype=np.float32), sitk.sitkFloat32)
```

Comment:
Does `list(target_shape) + [1]` means one more channel?

Reply:
Yes. `sitk.GetImageFromArray` needs to take the array with channel dimension as input defaultly.

By the way, since the type of `target_shape` is float, we need to cast to `np.int32` in advance to meet the type of `np.ones`'s argument.

Reply:
I see.